api/src/libs/bot-message.ts

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright © 2020 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+'use strict';
+
+import { logger } from '@bcgov/common-nodejs-utils';
+import DataManager from '../db';
+import { BotMessage } from '../db/model/request';
+import { NatsContext } from '../types';
+import shared from './shared';
+
+const dm = new DataManager(shared.pgPool);
+const { RequestModel, NamespaceModel } = dm;
+
+export const createBotMessageSet = async ( requestId: number, natsSubject: string, natsContext: NatsContext ): Promise<any> => {
+  if (!requestId) {
+      throw new Error('Cant get profile id');
+  }
+
+  const clusters = await NamespaceModel.findClustersForProfile(natsContext.profileId);
+  const promises: any = [];
+  clusters.forEach(cluster => {
```

Comment:
I've done this line quite a few times. I tend to use `for of` now as it allows for embedded await where as `for in` does not. I suppose they may have subtle differences with concurrency. Just say'n.

```JavaScript
for (const x of blurb) {
   await bla.doSometing(x);
}
````

Reply:
Valid observation. I have adjusted the use within this PR. Further refactoring could be done within the Registry to make this use universal.